cmake/HPX_AddModule.cmake

```diff
@@ -267,10 +267,33 @@ function(add_hpx_module libname modulename)
     # cmake-format: on
   endif()
 
-  target_link_libraries(
-    hpx_${modulename} PUBLIC ${${modulename}_MODULE_DEPENDENCIES}
-  )
-  target_link_libraries(hpx_${modulename} PUBLIC ${${modulename}_DEPENDENCIES})
+  if(${modulename}_MODULE_DEPENDENCIES)
+    # verify that all dependencies are from the same module category
+    foreach(dep ${${modulename}_MODULE_DEPENDENCIES})
```

Comment:
Should we make it optional, and only enable the check on CI? I'd guess it's not terribly slow, but future similar checks may add up...

Reply:
How do I do that? Do we have a flag already in place?

Reply:
No flag yet. We've done these kind of checks separately using shell scripts so far. What I had in mind was just adding an `ADVANCED` flag for this purpose.